### ISSUE TITLE
Disable styler for knitr R code chunk

### ISSUE THREAD

**Issue by gadenbuie** [r-lib/styler#781]
It would be nice to be able to disable styler for a particular R chunk in R Markdown documents without including `# styler: off` in the R code chunk.

Instead of this:

````markdown
```{r}
# styler: off
don_t_style_me <- c(____)
# styler: on
```
````

styler could be disabled with the [`tidy` chunk option](https://yihui.org/knitr/options/#code-decoration):

````markdown
```{r tidy=FALSE}
don_t_style_me <- c(____)
```
````

I'd recommend only disabling styler for the chunk if `tidy=FALSE`.

**Comment by lorenzwalthert** [comment]
I like the idea but I’d like to understand more what reasons there are to not style a chunk at all. Because that might offer more insight as to how we could improve styler further.

**Comment by gadenbuie** [comment]
Relatedly, if you happen to know of a way where we could trick the parser into accepting `____` as valid R code, that would be very useful (but I'm assuming that would be essentially impossible).

**Comment by lorenzwalthert** [comment]
FWIW: 

* There was a previous learnr related issue:  https://github.com/r-lib/styler/issues/657
* There is a postponed issue ok styling syntactically invalid code: #346 
* Similarly to #732, we could treat R markdown.

**Comment by gadenbuie** [comment]
That's helpful, thank you. I think the learnr-related issue you mentioned would be resolved by something like this. I also see my proposal as broadly helpful for R Markdown documents, since there are many motivating cases where an author might want to opt out of styling for one or more specific chunks without adding comments to the visible output or without having invalid code prohibit styling.

BTW, as an alternative suggestion, `styler="off"` might also be a reasonable chunk option if you want to avoid existing knitr chunk options.

**Comment by lorenzwalthert** [comment]
I see the point, but then we should prioritise this issue lower than the other open ones, in particular #346. Also, `tidy = FALSE` seems fine, I don't think we need to be able to distinguish between styling during knitting and styling when a file is styled with styler.

**Comment by lorenzwalthert** [comment]
Regarding the parser

> Relatedly, if you happen to know of a way where we could trick the parser into accepting ____ as valid R code, that would be very useful (but I'm assuming that would be essentially impossible).

I dont' know. You could use regex to replace code like `___` before parsing but it's brittle since I can also have 
```r
1 + "____"
```
which is valid R code.


Anyways, I hope you can use the strategy you initially proposed with `tidy = FALSE`. Also, it would be cool if it made sense to  incorporate this new feature of {styler} into the {learnr} docs somehow (maybe merge the PR only once this hits CRAN), because I am sure other people had this problem before. In particular it came up here in #657.
